internal/system/snapd_client.go

```diff
@@ -0,0 +1,155 @@
+package system
```

Comment:
I'd prefer this was in a separate package, `internal/snapd` (or `internal/snapdclient`), as it's completely independent of the rest of the system package, and it keeps the interface cleaner.